tools/CampaignConfigurationCheck/CampaignName.mjs

```diff
@@ -0,0 +1,15 @@
+import chalk from 'chalk';
+
+export default class CampaignName {
+	/**
+	 * @param {string} campaignName
+	 */
+	constructor( campaignName ) {
+		const testNumberRegexResult = campaignName.match( /_(\d{2})/ );
```

Comment:
```suggestion
		const testNumberRegexResult = campaignName.match( /_(\d{2})$/ );
```

You could also do a `split` and use the parts, to make this class more applicable for future uses and would allow for more thorough checking.

```
const campaignParts = campaignName.trim().split( '_' )`;
if ( campaignParts.length < 5 ) {
   throw new Error( `Campaign name '${campaignName}' does not follow pattern, must have at least 5 parts, separated with underscores` );
}
const campaignNumber = campaignParts.slice( -1 );
if ( !/^\d{2}$/.test(campaignNumber) ) {
    throw new Error( `Cannot parse test number "${ campaignNumber }" of campaign "${ campaignName }". Test number must be a zero-padded, 2-digit numeric string` );
}
// const campaignYear = campaignParts[0].substr(1);
// etc ...
```

Reply:
split certainly has its usecase, but if I start checking that thoroughly I see myself checking each element that split returns (to make sure I really got the right one) and nobody added a "-2-" for a test re-run for example.
From my perspective this would have a very strict assumption on the format.